pkg/backends/jmeter/jmeter.go

```diff
@@ -53,243 +41,35 @@ func New(
-				c.loadTest.Status.Phase = loadTestV1.LoadTestFinished
-				return nil
-			}
-
-			for _, containerStatus := range pod.Status.ContainerStatuses {
-				if containerStatus.State.Waiting == nil {
-					c.loadTest.Status.Phase = loadTestV1.LoadTestErrored
-					return nil
-				}
-				if containerStatus.State.Waiting.Reason != "Pending" &&
-					containerStatus.State.Waiting.Reason != "ContainerCreating" &&
-					containerStatus.State.Waiting.Reason != "PodInitializing" {
-					c.logger.Info(
-						"One of containers is unhealthy, marking LoadTest as errored",
-						zap.String("LoadTest", c.loadTest.GetName()),
-						zap.String("pod", pod.Name),
-						zap.String("namespace", namespace.GetName()),
-					)
-					c.loadTest.Status.Phase = loadTestV1.LoadTestErrored
-					return nil
-				}
-			}
-
-			return nil
-		}
-	}
-
-	job, err := c.kubeClientSet.BatchV1().Jobs(namespace.GetName()).Get(ctx, loadTestJobName, metaV1.GetOptions{})
-	if err != nil {
-		// The LoadTest resource may no longer exist, in which case we stop
-		// processing.
-		if errors.IsNotFound(err) {
-			_, err = c.kubeClientSet.BatchV1().Jobs(namespace.GetName()).Create(
-				ctx,
-				c.NewJMeterMasterJob(c.reportURL, c.podAnnotations),
-				metaV1.CreateOptions{},
-			)
-			return err
-		}
-		return err
-	}
-
-	// Get jmeter job in namespace and update the LoadTest status with
-	// the Job status
-	c.loadTest.Status.Phase = getLoadTestPhaseFromJob(job.Status)
-	c.loadTest.Status.JobStatus = job.Status
-
-	return nil
-}
-
-func (c *JMeter) createPodsWithTestdata(ctx context.Context, configMaps []*coreV1.ConfigMap, loadtest *loadTestV1.LoadTest, namespace string) error {
-	for i, cm := range configMaps {
-		configMap, err := c.kubeClientSet.CoreV1().ConfigMaps(namespace).Create(ctx, cm, metaV1.CreateOptions{})
-		if err != nil {
-			c.logger.Error("Error on creating testdata configMaps", zap.Error(err))
-			return err
-		}
-
-		_, err = c.kubeClientSet.CoreV1().Pods(namespace).Create(ctx, c.NewPod(i, configMap, c.podAnnotations), metaV1.CreateOptions{})
-		if err != nil {
-			c.logger.Error("Error on creating distributed pods", zap.Error(err))
-			return err
-		}
-	}
-	c.logger.Info("Created pods with test data", zap.String("LoadTest", loadtest.GetName()), zap.String("namespace", namespace))
-	return nil
-}
-
-func workerPodHasTimeout(startTime *metaV1.Time, loadtestStatus loadTestV1.LoadTestStatus) bool {
-	if startTime == nil {
-		return false
-	}
-
-	return time.Since(startTime.Time) > MaxWaitTimeForPods &&
-		loadtestStatus.Phase == loadTestV1.LoadTestCreating
-}
-
-func getLoadTestPhaseFromJob(status batchV1.JobStatus) loadTestV1.LoadTestPhase {
-	if status.Active > 0 {
-		return loadTestV1.LoadTestRunning
-	}
-
-	if status.Succeeded == 0 && status.Failed == 0 {
-		return loadTestV1.LoadTestStarting
-	}
-
-	return loadTestV1.LoadTestFinished
+	return c.backend.SyncStatus(ctx, *c.loadTest, &c.loadTest.Status)
```

Comment:
I would also add debug logging on each `CheckOrUpdateStatus` or we can move debug log to `SyncStatus` method.

Reply:
which log are you referring to? I didn’t remove any log, or did?